javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -631,6 +631,48 @@ public static boolean isUseOfElement(ExpressionTree node) {
         }
     }
 
+    /**
+     * Returns true if {@code tree} has a synthetic argument.
+     *
+     * <p>For an anonymous class is a subtype of an inner class and the creation of the inner class
+     * has an explicit receiver, the receiver is passed as the first argument of the constructor.
+     *
+     * <p>For example, <code>this.new MyInnerClass(){}</code> is changed to the following:
```

Comment:
What agent makes the change?  This method ought to be pure.

Nit: `{@code ...}` is considered better style than `<code>...</code>` and does work even though there are curly braces enclosed within.

Reply:
(Same questions about comments inline; is the duplication necessary?)